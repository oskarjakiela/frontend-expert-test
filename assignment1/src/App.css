.App {
  display: flex;
  height: 100vh;
  overflow: hidden;
}

.App__aside {
  border-right: 1px solid rgba(0, 0, 0, .1);
  width: 20%;
  overflow-y: scroll;
}

.App__main {
  width: 80%;
}

@media all and (max-width: 736px) {
  .App {
    flex-direction: column;
  }

  .App__aside,
  .App__main {
    width: 100%;
  }

  .App__aside {
    order: 1;
  }

  .App__main {
    order: 0;
  }
}