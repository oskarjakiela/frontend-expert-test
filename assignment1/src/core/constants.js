export const API_URL = 'https://www.deskbookers.com/nl-nl/search/ajax.json';
