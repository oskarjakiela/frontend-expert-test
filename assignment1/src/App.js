import axios from 'axios';
import { compose, map, pathOr, pick } from 'ramda';
import React, { Component } from 'react';

import './App.css';
import { API_URL } from './constants';


class App extends Component {
  componentDidMount() {
    axios.get(API_URL)
      .then(pick(['data']))
      .then((newState) => this.setState(newState));
  }

  render() {
    const renderResults = compose(
      map(({ id, name }) => <div key={id}>{ name }</div>),
      pathOr([], ['data', 'rows'])
    );

    return (
      <div className="App">
        { renderResults(this.state) }
      </div>
    );
  }
}

export default App;
