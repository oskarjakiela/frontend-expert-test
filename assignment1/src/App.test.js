import axios from 'axios'
import moxios from 'moxios'
import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';

import App from './App';
import { API_URL } from './constants';


describe('<App />', () => {
  const rows = [{
    id: 1,
    name: 'Foo 1',
    location_name: 'Bar',
    location_city: 'Wonderland',
  }, {
    id: 2,
    name: 'Foo 2',
    location_name: 'Bar',
    location_city: 'Wonderland',
  }];

  beforeEach(() => {
    moxios.install();

    moxios.stubRequest(API_URL, {
      status: 200,
      response: { rows },
    })
  });

  afterEach(() => {
    moxios.uninstall();
  });

  it('renders without crashing', (done) => {
    const div = document.createElement('div');

    render(<App />, div);

    moxios.wait(() => {
      unmountComponentAtNode(div);
      done();
    });
  });

  it('renders correctly', (done) => {
    const component = renderer.create(<App />)

    moxios.wait(() => {
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      done();
    });
  });

});
