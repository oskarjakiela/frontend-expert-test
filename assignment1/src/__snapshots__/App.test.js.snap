// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="App"
>
  <div>
    Foo
  </div>
  <div>
    Bar
  </div>
</div>
`;
